src/navx_node.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * navx_node.cpp
+ * Runs the Kauai Labs NavX, using modified NavX library
+ * VERSION: 0.0
+ * Last changed: 2019-04-01
+ * Authors: Jude Sauve <sauve031@umn.edu>
+ * Maintainers: Jude Sauve <sauve031@umn.edu>
+ * MIT License
+ * Copyright (c) 2018 GOFIRST-Robotics
+ */
+
+// ROS Libs
+#include <ros/ros.h>
+#include <sensor_msgs/Imu.h>
+
+// Native_Libs
+#include <string>
+
+// Custom_Libs
+#include "ahrs/AHRS.h"
+
+// Subscribers (inputs)
+//    update_timer (Timer)
+//      Update loop for reading / querying IMU
+
+// Publishers (outputs)
+//    imu_pub (sensor_msgs/Imu): imu/data
+//      The published imu data
+
+// Parameters (settings)
+//    frequency (double): default=50.0
+//      The update frequency of the update loop
+
+
+// ROS Node and Publishers
+ros::NodeHandle * nh;
+ros::NodeHandle * pnh;
+ros::Publisher imu_pub;
+
+// ROS Callbacks
+void update_callback(const ros::TimerEvent&);
+
+// ROS Params
+double frequency = 50.0;
+
+// Global_Vars
+AHRS* com;
+
+int main(int argc, char** argv) {
+  // Init IMU
+  com = new AHRS("/dev/ttyACM0");
```

Comment:
The proper order in main() is`{
// ROS Init
// Params
// Init (other stuff)
// Subscribers
// Publishers
// Spin
// Close/Delete any non-handled resources
}`
The order is because we need params usually for init'ing other stuff, and then subs before pubs because it matches the info flow: inputs to outputs. 
We don't have to clean up com after spin, b/c it's handled. But it should be after params, b/c params might set the device path/name, instead of hardcoding it (likewise for frequency - do you see how the timer is made with the default frequency instead of the parameter value?)